tests/zquantum/core/estimation/estimation_test.py

```diff
@@ -603,11 +629,62 @@ def test_evaluate_non_measured_estimation_tasks_fails_with_non_zero_shots(
             ),
         ],
         [
-            ExpectationValues(np.array([0])),
-            ExpectationValues(np.array([-2 * (np.cos(np.pi / 8) ** 2 - 0.5) * 2])),
+            ExpectationValues(np.array([0]), [np.array([[1]])], [np.array([[0.001]])]),
+            ExpectationValues(
+                np.array([-2 * (np.cos(np.pi / 8) ** 2 - 0.5) * 2]),
+                [np.array([[4]])],
+                [np.array([[0.002]])],
+            ),
         ],
     ),
 ]
+# When number_of_shots is high, the covariance will tend to zero. So we need extra
+# test cases to ensure covariance is being caluclated correctly.
+TEST_CASES_NONEIGENSTATES_WITH_LOW_NUMBER_OF_SHOTS = [
+    (
+        4
+        * [
+            EstimationTask(
+                IsingOperator("[Z0] + [Z1]"),
+                circuit=Circuit([X(0), X(1)]),
+                number_of_shots=10,
+            ),
+        ],
+        [
+            ExpectationValues(
+                np.array([0]),
+                [np.array([[1, 0], [0, 1]])],
+                [np.array([[0, 0], [0, 0.5]])],
+            ),
+            ExpectationValues(
+                np.array([0]),
+                [np.array([[1, 0], [0, 1]])],
+                [np.array([[0.5, 0], [0, 0]])],
+            ),
+            ExpectationValues(
+                np.array([0]),
+                [np.array([[1, -1], [-1, 1]])],
+                [np.array([[0.5, -0.5], [-0.5, 0.5]])],
+            ),
+            ExpectationValues(
+                np.array([0]),
+                [np.array([[1, 1], [1, 1]])],
+                [np.array([[0.05, 0.05], [0.05, 0.05]])],
+            ),
+        ],
+    ),
+]
+
+
+# needs it's own class otherwise issues arise with calling run_circuitset_and_measure.
+class MockBackendForTestingCovariancewhenNumberOfShotsIsLow:
+    def run_circuitset_and_measure(self, circuit, shots_per_circuit):
+        return [
+            Measurements([(0, 1), (0, 0)]),
+            Measurements([(1, 0), (0, 0)]),
+            Measurements([(1, 0), (0, 1)]),
+            Measurements([(1, 1), (0, 0)] * 10),
```

Comment:
Are these values closely tied to the values in matrices in `TEST_CASES_NONEIGENSTATES_WITH_LOW_NUMBER_OF_SHOTS`? If so, would it make sense to treat them as the "inputs" to your unit test?

Reply:
I'm not sure how to make these outputs for `run_circuit_and_measure` while still putting them as inputs to the test.

Reply:
[not a blocker] How about hardcoding those right inside the test method or under a [`pytest.fixture`](https://docs.pytest.org/en/6.2.x/fixture.html)?